apps/tailwind-components/components/input/Ref.vue

```diff
@@ -0,0 +1,272 @@
+  offset.value = 0;
+  searchTerms.value = newSearchTerms;
+  loadOptions({ limit: props.limit, searchTerms: searchTerms.value });
+}
+
+function select(label: string) {
+  console.log("select " + label);
+  if (!props.isArray) {
+    selectionMap.value = {};
+  }
+  selectionMap.value[label] = optionMap.value[label];
+  emit(
+    "update:modelValue",
+    props.isArray
+      ? Object.values(selectionMap.value).map((value) =>
+          extractPrimaryKey(value)
+        )
+      : extractPrimaryKey(optionMap.value[label])
+  );
+}
+
+function extractPrimaryKey(value: any) {
+  const result = {} as columnValueObject;
+  tableMetadata.value.columns
+    .filter((column) => column.key === 1)
+    .forEach((column) => {
+      result[column.id] = value[column.id];
+    });
+  return result;
+}
+
+function deselect(label: string) {
+  delete selectionMap.value[label];
+  emit(
+    "update:modelValue",
+    props.isArray
+      ? Object.values(selectionMap.value).map((value) =>
+          extractPrimaryKey(value)
+        )
+      : undefined
+  );
+}
+
+function clearSelection() {
+  selectionMap.value = {};
+  emit("update:modelValue", props.isArray ? [] : undefined);
+  updateSearch(""); //reset
+}
+
+function loadMore() {
+  offset.value += props.limit;
+  loadOptions({
+    offset: offset.value,
+    limit: props.limit,
+    searchTerms: searchTerms.value,
+  });
+}
+</script>
+
+<template>
+  <div
+    class="flex flex-wrap gap-2 mb-2"
+    v-if="isArray ? selection.length : selection"
+  >
+    <Button
+      v-for="label in isArray ? selection : [selection]"
+      icon="cross"
+      iconPosition="right"
+      type="filterWell"
+      size="tiny"
+      @click="deselect(label as string)"
+    >
+      {{ label }}
+    </Button>
+  </div>
+  <div class="flex flex-wrap gap-2 mb-2">
+    <ButtonText
+      @click="toggleSearch"
+      :inverted="inverted"
+      :aria-controls="`search-for-${id}`"
+    >
+      Search
+    </ButtonText>
+    <ButtonText @click="clearSelection" :inverted="inverted">
+      Clear all
+    </ButtonText>
+  </div>
+  <template v-if="showSearch && initialCount > limit">
+    <InputLabel :for="`search-for-${id}`" class="sr-only">
+      search in {{ columnName }}
+    </InputLabel>
+    <InputSearch
+      :id="`search-for-${id}`"
+      :modelValue="searchTerms"
+      @update:modelValue="updateSearch"
+      class="mb-2"
+      :placeholder="`Search in ${columnName}`"
+      :aria-hidden="!showSearch"
+    />
+  </template>
+  <template v-if="!hasNoResults">
+    <InputCheckboxGroup
+      v-if="isArray"
+      :id="id"
+      :options="listOptions"
+      :modelValue="(selection as string[])"
+      @select="select"
+      @deselect="deselect"
+      :inverted="inverted"
+    />
+    <InputRadioGroup
+      v-else
+      :id="id"
+      :options="listOptions"
+      :modelValue="(selection as string)"
+      @select="select"
+      @deselect="deselect"
+      :inverted="inverted"
+    />
+    <ButtonText
```

Comment:
Should we limit the number of visible options or put this into a scrollable area? In the complex form example, I continuously click the "load 10 more" button. The number of visible options becomes quite long and there is content shift. It might be better to place this in a scrollable area to prevent forms from growing in size. I might also make the search visible by default in this case.